code/modules/mob/living/basic/lavaland/basilisk/basilisk.dm

```diff
@@ -0,0 +1,85 @@
+/// Watchers' ground-dwelling cousins, they shoot at you until they get into melee and absorb laser fire to power up.
+/mob/living/basic/mining/basilisk
+	name = "basilisk"
+	desc = "A territorial beast, covered in a diamond shell which absorbs heat. Its stare causes victims to freeze from the inside."
+	icon_state = "Basilisk"
+	icon_living = "Basilisk"
+	icon_dead = "Basilisk_dead"
+	speak_emote = list("chimes")
+	damage_coeff = list(BRUTE = 1, BURN = 0.1, TOX = 1, CLONE = 1, STAMINA = 0, OXY = 1)
+	speed = 20
+	maxHealth = 200
+	health = 200
+	obj_damage = 60
+	melee_damage_lower = 12
+	melee_damage_upper = 12
+	attack_verb_continuous = "bites into"
+	attack_verb_simple = "bite into"
+	throw_blocked_message = "bounces off the shell of"
+	attack_sound = 'sound/weapons/bladeslice.ogg'
+	attack_vis_effect = ATTACK_EFFECT_BITE
+	ai_controller = /datum/ai_controller/basic_controller/Basilisk
+	butcher_results = list(
+		/obj/item/stack/sheet/bone = 1,
+		/obj/item/stack/ore/diamond = 2,
+		/obj/item/stack/sheet/sinew = 2,
+	)
+	/// The component we use for making ranged attacks
+	var/datum/component/ranged_attacks/ranged_attacks
+
+/mob/living/basic/mining/basilisk/Initialize(mapload)
+	. = ..()
+	AddComponent(/datum/component/basic_mob_attack_telegraph)
+	ranged_attacks = AddComponent(/datum/component/ranged_attacks, projectile_type = /obj/projectile/temp/watcher, projectile_sound = 'sound/weapons/pierce.ogg')
+
+/mob/living/basic/mining/basilisk/Destroy()
+	QDEL_NULL(ranged_attacks)
+	. = ..()
+
+/mob/living/basic/mining/basilisk/welder_act(mob/living/user, obj/item/tool)
+	. = ..()
+	heat_up() // Who would do this?
+
+/mob/living/basic/mining/basilisk/bullet_act(obj/projectile/bullet, def_zone, piercing_hit)
+	. = .. ()
+	if (istype(bullet, /obj/projectile/temp))
+		var/obj/projectile/temp/heat_bullet = bullet
+		if (heat_bullet.temperature < 0)
+			return
+		heat_up()
+		return
+
+	if (bullet.damage == 0 || bullet.damage_type != BURN)
+		return
+	heat_up()
+
+/mob/living/basic/mining/basilisk/Moved(atom/old_loc, movement_dir, forced, list/old_locs, momentum_change)
```

Comment:
ftr i prefer using signals when it's something optional to the proc (as in going onto lava doesn't really affect the whole Move/Moved chain) but it's ultimately whatever given how other maintainers feel about it